src/lib/input/input.ts

```diff
@@ -90,14 +95,46 @@ export class MdHint {
   @Input() align: 'start' | 'end' = 'start';
 }
 
+/**
+ * Enables auto expanding behaviour for the textarea
+ */
+@Directive({
+  selector: 'textarea[mdAutosize]'
+})
+export class MdAutosize implements OnInit {
+  @HostListener('input', ['$event.target'])
+  public onChange() {
```

Comment:
It's what the `@HostListener` is calling. Or you mean it should be private? I suppose..